chia/rpc/wallet_rpc_api.py

```diff
@@ -1228,15 +1228,28 @@ async def verify_signature(self, request) -> EndpointResult:
         :param request:
         :return:
         """
+        input_message: str = request["message"]
+        signing_mode: Optional[str] = request.get("signing_mode", None)
+        if signing_mode == "chip_0002":
+            # CHIP-0002 message signatures are made over the tree hash of:
+            #   ("Chia Signed Message", message)
+            message_to_verify: bytes = Program.to((CHIP_0002_SIGN_MESSAGE_PREFIX, input_message)).get_tree_hash()
+        else:
+            # Signed message is expected to be a hex string
+            message_to_verify = hexstr_to_bytes(input_message)
```

Comment:
Added a SigningMode enum to handle the CHIP-0002 case as well as the default case (now denoted as `SigningMode.BLS_MESSAGE_AUGMENTATION_HEX_INPUT`). A third case is also handled when dealing with a UTF-8 message `SigningMode.BLS_MESSAGE_AUGMENTATION_UTF8_INPUT`.

I've left out the `else:` branch because an instance of SigningMode is constructed from a string, raising an exception if the string is not a valid signing mode value. Including the `else:` branch would lead to unreachable code in this case.